### PR TITLE
Updating Android Native Checkout Version to 1.2.0

### PR DESCRIPTION
**Pull request by chpypl**
Thank you for your contribution to PayPal. 

> Before submitting this PR, note that we cannot accept language translation PRs. We have a dedicated localization team to provide the translations. If there is an error in a specific translation, you may open an issue and we will escalate it to the localization team.

### Summary of changes

- Updating the Android Native Checkout SDK to 1.2.0
- Adding in the auth config for nxo to be able to set the merchant passed email

 ### Checklist

 - [x] Added a changelog entry

-

### CODE REVIEW

**Review by sshropshire — APPROVED** [review]

**Review by warmkesselj — APPROVED** [review]

**Review by jaxdesmarais — CHANGES_REQUESTED** [review]
Can we update properties and the changelog to match the BT android PR: https://github.com/braintree/braintree_android/pull/829

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # PayPal Android SDK Release Notes
 
+## unreleased
+
+* PayPalNativeCheckout
+  * Bump native-checkout version to release `1.2.0`
+  * Add `email` to `PayPalNativeCheckoutRequest`
```
Comment (jaxdesmarais) [inline CHANGELOG.md]:
```suggestion
  * Add `userAuthenticationEmail` to `PayPalNativeCheckoutRequest`
```

File: PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClient.kt [hunk 2]
```diff
@@ -83,14 +84,16 @@ class PayPalNativeCheckoutClient internal constructor(
         orderId = request.orderId
         CoroutineScope(dispatcher).launch(exceptionHandler) {
             try {
+                val authConfig: AuthConfig? = request.email?.let { AuthConfig(it) }
```
Comment (jaxdesmarais) [inline PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClient.kt]:
```suggestion
                val authConfig: AuthConfig? = request.userAuthenticationEmail?.let { AuthConfig(it) }
```

File: PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutRequest.kt [hunk 3]
```diff
@@ -7,5 +7,10 @@ data class PayPalNativeCheckoutRequest(
     /**
      * The order ID associated with the request.
      */
-    val orderId: String
+    val orderId: String,
+
+    /**
+     * Optional: The email for preloading authentication
+     */
+    val email: String? = null
```
Comment (jaxdesmarais) [inline PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutRequest.kt]:
```suggestion
    val userAuthenticationEmail: String? = null
```

File: PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutRequest.kt [hunk 4]
```diff
@@ -7,5 +7,10 @@ data class PayPalNativeCheckoutRequest(
     /**
      * The order ID associated with the request.
      */
-    val orderId: String
+    val orderId: String,
+
+    /**
+     * Optional: The email for preloading authentication
```
Comment (jaxdesmarais) [inline PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutRequest.kt]:
```suggestion
     * Optional: User email to initiate a quicker authentication flow in cases where the user has a PayPal Account with the same email.
```

**Review by jaxdesmarais — APPROVED** [review]

**Pull request merged**